### ISSUE TITLE
Logging sssd output to `/var/log/sssd` instead of stderr?

### ISSUE THREAD

**Issue by pat-s** [rstudio/rstudio-docker-products#235]
I was wondering if it would be possible to redirect the `sssd` log output to  `/var/log/sssd/<domain name_log` instead of stderr by default?

https://github.com/rstudio/rstudio-docker-products/blob/659c1d25b16c3d4070549b301f1cbad4c42792c2/workbench/startup-user-provisioning/sssd.conf#L3

This could be achieved by starting the service via `--logger=files` instead of `--logger=stderr`.

**Comment by colearendt** [comment]
Woops - missed this one!

So the downside of writing to `/var/log` within a container is that the container's ephemeral filesystem will fill up sooner-or-later. This is particularly problematic in systems like Kubernetes, which will eventually kill the pod for using too much ephemeral storage.

This is possible to be customized by mounting a new `sssd.conf` file on top of the default at `/startup/user-provisioning/sssd.conf`.

Unfortunately, I think the way we have set things up in the helm chart (which may be where you would need to mount this?) there is not a fantastic solution - would a way to overwrite this file in the helm chart be helpful for your setup?

https://github.com/rstudio/helm/blob/1e67050ac2e328a1707193ef418b91ce142c4c34/charts/rstudio-workbench/templates/configmap-startup.yaml#L44

**Comment by pat-s** [comment]
> This is particularly problematic in systems like Kubernetes, which will eventually kill the pod for using too much ephemeral storage.

I see! I haven't though about this one yet tbh as usually user data (both for RSW and RSC) are way higher than log size in practice - which is why we needed to raise ephemeral storage limits in our pods anyhow.

We're building our own images which we use both for normal docker deployments and in k8s. We use file logging as a default, primarily for normal docker deployments. Persisting log dirs from the container helps us to better understand possible container issues in post-hoc analyses. So far we haven't seen any substantial storage issues in k8s but we'll keep investigating.

Overall this discussion applies to any kind of logging and not just `sssd`. Maybe adding a sentence about the possible options that admins have and why the default is set as it is (referring your comment from above) could help? Just an idea though :)

**Comment by colearendt** [comment]
Yes, that's a great call! It is worth noting - if you are mounting a persistent directory to store the logs in, then you have solved the ephemeral storage problem 😄 It is also one of the reasons that our documentation highlights the importance of mounting the home directory / data directory / etc. for the various services - because then that shared storage is written to a volume / persistent storage and does not take up ephemeral storage space.

We will try to find a space to make the docs more clear on this front. One thing that we are working to is a consistent set of logging specifications across our products - so that it will be possible to write all logs to a file based on product config / without changing the entrypoint script / etc. We will shoot for having examples of that too!

**Comment by colearendt** [comment]
Closing this! Please let us know if you have any issues getting this to work for you!
